### ISSUE TITLE
How can I avoid obtaining loss: nan - acc: 0.0000e+00 for CNN-LSTM in keras?

### ISSUE THREAD

**Issue by oytungunes** [oytungunes/Thyroid-classification-with-Decision-Trees#1]
Hello everyone,

I am trying to make 3 class classification with CNN-LSTM, I have prepared my data on MATLAB, and train with the following code. It works for binary classification, where I used binary cross entropy, but does not work for multi class case I used sparse_categorical_crossentropy. I am having nan for loss and 0.0000e+00 for accuracy. What is wrong here?
Thanks for the help.

```python


import scipy.io
import numpy as np
mat = scipy.io.loadmat('frames3class_train80_test20.mat')
XTrain = mat['tensortrain']
y = mat['YTrain']

##
trainlength=2400
X = np.zeros(shape=(trainlength,8,18,400,1))
X[:,:,:,:,0]=XTrain


import keras
from keras.layers import Dense, LSTM, Flatten, TimeDistributed, Conv2D, Dropout,MaxPooling2D
from keras.models import Sequential
width = 18
height = 400
kernelsize =5 
numberOfHiddenUnits =50 

model = Sequential()
model.add(TimeDistributed(Conv2D(2, (kernelsize, kernelsize), activation='relu'),
                          input_shape=(None, width, height, 1)))
model.add(TimeDistributed(MaxPooling2D(pool_size=(2, 2))))
model.add(TimeDistributed(Flatten()))
model.add(LSTM(numberOfHiddenUnits))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss=keras.losses.sparse_categorical_crossentropy, optimizer='adam', metrics=['acc'])

model.fit(X, y,batch_size=16,epochs=30)

```

Result on the console is :

```python
 16/2400 [..............................] - ETA: 4s - loss: nan - acc: 0.0000e+00
  80/2400 [>.............................] - ETA: 2s - loss: nan - acc: 0.0000e+00
 144/2400 [>.............................] - ETA: 2s - loss: nan - acc: 0.0000e+00
 192/2400 [=>............................] - ETA: 2s - loss: nan - acc: 0.0000e+00
 256/2400 [==>...........................] - ETA: 2s - loss: nan - acc: 0.0000e+00
 320/2400 [===>..........................] - ETA: 2s - loss: nan - acc: 0.0000e+00
```

**Comment by oytungunes** [comment]
Ok I changed the labels with one hot encoding, and the loss function accordingly it is solved.
